libs/sdk-ui-dashboard/src/model/commandHandlers/filterContext/changeFilterContextSelectionHandler.ts

```diff
@@ -93,18 +101,55 @@ export function* changeFilterContextSelectionHandler(
     yield call(dispatchFilterContextChanged, ctx, cmd);
 }
 
+function loadAttribute(attributeRef: ObjRef, ctx: DashboardContext) {
+    return ctx.backend.workspace(ctx.workspace).attributes().getAttribute(attributeRef);
+}
+
+function loadAttributeDisplayForm(displayFormRef: ObjRef, ctx: DashboardContext) {
+    return ctx.backend.workspace(ctx.workspace).attributes().getAttributeDisplayForm(displayFormRef);
+}
+
 function* getAttributeFiltersUpdateActions(
     attributeFilters: IDashboardAttributeFilter[],
     resetOthers: boolean,
+    ctx: DashboardContext,
 ): SagaIterator<AnyAction[]> {
     const updateActions: AnyAction[] = [];
     const handledLocalIds = new Set<string>();
 
     for (const attributeFilter of attributeFilters) {
         const filterRef = attributeFilter.attributeFilter.displayForm;
-        const dashboardFilter: ReturnType<
-            ReturnType<typeof selectFilterContextAttributeFilterByDisplayForm>
-        > = yield select(selectFilterContextAttributeFilterByDisplayForm(filterRef));
+        let dashboardFilter: ReturnType<ReturnType<typeof selectFilterContextAttributeFilterByDisplayForm>> =
+            yield select(selectFilterContextAttributeFilterByDisplayForm(filterRef));
+
+        if (!dashboardFilter) {
+            if (isUriRef(filterRef) && !ctx.backend.capabilities.supportsObjectUris) {
+                throw new NotSupported(
+                    "Unsupported filter ObjRef! Please provide IdentifierRef instead of UriRef.",
+                );
+            }
+
+            const filterDF: PromiseFnReturnType<typeof loadAttributeDisplayForm> = yield call(
```

Comment:
i think we have some resolvers/code in `model/utils` that may help out with this. the idea is.. in 99% the info about display form is available in catalog (which now loads all DFs) so we have to go down to backend only in corner cases.